### PR TITLE
Features/provide support for stacking multiple providers

### PR DESCRIPTION
**Pull request by kwesolowski**
Added mechanism to allow integration of geoid undulation with SRTM data.

### CODE REVIEW

File: src/NSrtm.Core/Level.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace NSrtm.Core
+{
+    public enum Level
+    {
+        EllipsoidWgs84,
+        GeoidEgm2008,
+        Terrain
```
Comment (Ciszq) [inline src/NSrtm.Core/Level.cs:7]: AFAIK Egm supports multiple formats not only 2008.

Wgs 84 is a ellipsoid with specific radius, so name EllipsoidWgs84 is reduplication. Same with Egm.

File: src/NSrtm.Core/IElevationProvider.cs [hunk 2]
```diff
@@ -33,5 +23,15 @@ public interface IElevationProvider
         /// <param name="longitude"></param>
         /// <returns></returns>
         Task<double> GetElevationAsync(double latitude, double longitude);
+
```
Comment (Ciszq) [inline src/NSrtm.Core/IElevationProvider.cs]: Why these prop are empty?

File: src/NSrtm.Core/IElevationProvider.cs [hunk 3]
```diff
@@ -4,21 +4,11 @@
 namespace NSrtm.Core
 {
     /// <summary>
-    ///     Provides elevation of ground for WGS84 locations.
+    ///     Provides elevation for WGS84 locations.
```
Comment (Ciszq) [inline src/NSrtm.Core/IElevationProvider.cs:7]: WGS84? I thought it was also the interface for EGM

File: src/NSrtm.Core/Level.cs (hunk 1 above)
Comment (kwesolowski) [inline src/NSrtm.Core/Level.cs:7]: I am aware, but such naming makes it much easier to reason about code, and facts you mentioned are not obvious for all team members - so such little redundancy is not problem.

Please fill extra `Levels` when it is required (feel free to update this branch).

File: src/NSrtm.Core/IElevationProvider.cs (hunk 3 above)
Comment (kwesolowski) [inline src/NSrtm.Core/IElevationProvider.cs:7]: Yeah, but we must standardize what coordinates are used - can you suggest better name?

File: src/NSrtm.Core/Composer/SubtractingComposerElevationProvider.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            double negative = _providerNegative.GetElevation(latitude, longitude);
+            return positive - negative;
+        }
+
+        public async Task<double> GetElevationAsync(double latitude, double longitude)
+        {
+            var positiveTask = _providerPositive.GetElevationAsync(latitude, longitude);
+            var negativeTask = _providerNegative.GetElevationAsync(latitude, longitude);
+
+            await Task.WhenAll(positiveTask, negativeTask);
```
Comment (Ciszq) [inline src/NSrtm.Core/Composer/SubtractingComposerElevationProvider.cs]: Why you didn't use this https://msdn.microsoft.com/en-us/library/hh194766(v=vs.110).aspx?

**Pull request merged**